pkg/endpoint/bpf.go

```diff
@@ -179,15 +179,25 @@ func (e *Endpoint) addNewRedirectsFromDesiredPolicy(ingress bool, desiredRedirec
 			if !e.hasSidecarProxy || l4.L7Parser != policy.ParserTypeHTTP {
 				var finalizeFunc revert.FinalizeFunc
 				var revertFunc revert.RevertFunc
-				redirectPort, err, finalizeFunc, revertFunc = e.updateProxyRedirect(l4, proxyWaitGroup)
+
+				proxyID := ""
+				proxyID, err = e.ProxyID(e.desiredPolicy.NamedPortsMap, l4)
+				if err != nil {
+					// skip redirects for which a proxyID cannot be created.
+					// Right now this happens only due to named port mapping not existing and
+					// we'll be called again when the mapping is available.
+					log.WithError(err).WithField(logfields.EndpointID, e.ID).Debug("Skipping adding redirect")
```

Comment:
Shouldn't it Error instead of Debug?

Reply:
There are two scenarios under which this can happen:
1) Ingress policy refers to a port name that is not defined for the POD. In that case the redirect will not be added, but also the L4 policy allow will not be added to the bpf policy map, so traffic to the named port with an unknown port number will not be allowed. This is correct if the named port is missing due to POD actually not listening on that port. If the port name missing from the POD spec is an error, then traffic will be dropped, and maybe a Warning log message would be helpful in figuring out what is going on. IMO Error is too much.

2) Egress policy refers to a port name that is not defined. Egress policies are defined in terms of the destination port number, and typically the named port would be specified in the POD spec of the destination POD. If it is, it may not be propagated to the source node yet. In this case the policy will be regenerated after the port name mapping arrives. If no POD in the cluster defines the port name, then the traffic to that named port will be dropped. If the missing port name is due to no POD actually listening on that port then the behavior is correct. If the port name is missing from the (destination) POD spec by accident, then a Warning log message may be helpful.

=> Will change from Debug to Warning :-)